src/agents/factory.py

```
@@ -158,16 +166,26 @@ def _create_subagent(
         sub_internal_tools = self._filter_tools(
             internal_tool_dict, sub_internal_patterns, internal_module_map
         )
+
+        use_catalog = (
+            subagent_config.tools.use_catalog if subagent_config.tools else False
+        )
+        sub_llm_tools = sub_impl_tools + sub_mcp_tools + [think]
+        tools_in_catalog = []
+        if use_catalog:
+            tools_in_catalog = sub_impl_tools + sub_mcp_tools
+            sub_llm_tools = self.tool_factory.get_catalog_tools() + [think]
```

Ruff (0.14.5)

Consider `[*self.tool_factory.get_catalog_tools(), think]` instead of concatenation

Replace with `[*self.tool_factory.get_catalog_tools(), think]`

(RUF005)